### ISSUE TITLE
[urgent] lang tag for abstracts does not flow through for datasets

### ISSUE THREAD

**Issue by mietcls** [ugent-library/biblio-backoffice#1006]
# Bug

Language tag for abstracts does not flow through for datasets.

### Case: has 2 abstracts.
backoffice: https://backoffice.biblio.ugent.be/dataset/01GPFYQG4ADTSJWS70KYJ4SNDS
<img width="1054" alt="Scherm­afbeelding 2023-02-02 om 13 42 42" src="https://user-images.githubusercontent.com/1572885/216328023-fdf01cca-5fd2-486a-a4e1-62bdeb501f14.png">

MODS36 says "und"
https://biblio.ugent.be/oai?verb=GetRecord&identifier=01GPFYQG4ADTSJWS70KYJ4SNDS&metadataPrefix=mods_36
<img width="1360" alt="Scherm­afbeelding 2023-02-02 om 13 43 12" src="https://user-images.githubusercontent.com/1572885/216328127-991808a2-818f-4d07-8c25-5c274330db6b.png">

### Case: has 1 abstract
https://biblio.ugent.be/oai?verb=GetRecord&identifier=8708772&metadataPrefix=mods_36
![Image](https://user-images.githubusercontent.com/1572885/216391579-77cb9c05-7989-4cdf-a951-76e184b6a0d8.png)

### Expected result
`TBD` Language tag flows through : `en`

See page 100; FRIS expects `en` https://researchportal.be/sites/default/files/block-attachments/2022-08/Integration%20Guide%20FRIS%202.24_0.pdf

**Comment by nicolasfranck** [comment]
@mietcls this looks like an issue that was discussed earlier, right? See https://github.com/ugent-library/biblio-backoffice/issues/1002

**Comment by mietcls** [comment]
No, for this one we actually set a language in the back-office, so we can give that information. For keywords it is unfortunately missing. So for abstracts we can give the value without guessing. What information are we still missing to pass on the language @nicolasfranck?

**Comment by nicolasfranck** [comment]
right, that language is statically put in the [biblio code](https://github.ugent.be/Universiteitsbibliotheek/biblio/blob/master/views/export/_mods_36_body.tt#L102). It does not even inspect the language tag.

**Comment by nics** [comment]
@mietcls the frontend currently has no concept of abstract language, is this needed?

**Comment by mietcls** [comment]
@nics yes, if possible – it's a FRIS requirement for the datasets, and because we actually ask for the data it would be nice to share it (not just with FRIS). What is the impact, what changes does this entail?

**Comment by nicolasfranck** [comment]
@nics I could change the model definition of `abstract` in the biblio frontend from an `array of strings` to an `array of objects` like in biblio backoffice, but that requires a complete reindexation of the frontend index, a rewrite of the views (where they are used), so I do not know if I can do it without severely breaking anything.

**Comment by nics** [comment]
it's a bit more complicated than that, we have to be backwards compatible
some options:
- change it to an array of objects, but keep exposing it as an array of strings in all api's
- add an array with the language codes
- add a new array of objects alongside the old one

**Comment by nics** [comment]
see https://github.ugent.be/Universiteitsbibliotheek/biblio/commit/058454bb8404472b1d194b6a8b50c36c8691f8b1

**Comment by nics** [comment]
@mietcls abstract language is now set in frontoffice mods 3.6 (but only incrementaly when a record is updated, to update everything we n eed to do a full reindex of the frontoffice)

**Comment by nicolasfranck** [comment]
And now it does. Had to reindex because it depended on a new field, and the old index never picks up new fields.

Side effect, reported by Myriam and Stephanie: around updates 150K in GISMO.

**Comment by nicolasfranck** [comment]
@mietcls and now?

**Comment by mietcls** [comment]
The abstract language still shows "und".

Another possible related case: only records where I re-assigned UGent authors seem to have been added to the excel export. Could also be unrelated, but who knows. https://github.com/ugent-library/biblio-backoffice/issues/1067

**Comment by nics** [comment]
fixed by full reindex

**Comment by mietcls** [comment]
This one updated immediately: https://biblio.ugent.be/oai?verb=GetRecord&identifier=01GYCDM8GWQEAG9AY20CW8GR2V&metadataPrefix=mods_36

Hard to figure out which ones update immediately, and which ones do not.
